osu.Game/Overlays/ChatConsole.cs

```diff
@@ -0,0 +1,182 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Drawables;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
+using osu.Framework.Threading;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Chat;
+using osu.Game.Online.Chat.Display;
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Input;
+using OpenTK.Input;
+using osu.Framework;
+
+namespace osu.Game.Overlays
+{
+    public class ChatConsole : Container, IStateful<ChatConsoleState>
+    {
+        private APIAccess api => ((OsuGameBase)Game).API;

```

Comment:
No upward references. Constructor should be passed an APIAccess reference instead.